### ISSUE TITLE
Not able to install plugin `markdown-it-pseudocodejs`

### ISSUE THREAD

**Issue by jawardell** [samuelhorwitz/markdown-it-pseudocodejs#1]
Hello, I am trying to follow your steps for setting up `markdown-it-pseudocodejs`. I was able to run `npm install markdown-it` just fine, but when I try to install the plugin using `npm install markdown-it-pseudocodejs` like your instructions say, I get the following error.

```
machine:~ jwardell1$ npm install markdown-it-pseudocodejs
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/markdown-it-pseudocodejs - Not found
npm ERR! 404 
npm ERR! 404  'markdown-it-pseudocodejs@*' is not in this registry.
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jwardell1/.npm/_logs/2023-03-01T15_09_30_584Z-debug-0.log
```

I tried installing using your git url and got the following errors:
```
machine:~ jwardell1$ npm install https://github.com/samuelhorwitz/markdown-it-pseudocodejs
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: undefined@undefined
npm ERR! Found: markdown-it@13.0.1
npm ERR! node_modules/markdown-it
npm ERR!   markdown-it@"^13.0.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer markdown-it@"^6.0.0" from markdown-it-pseudocodejs@2.0.3
npm ERR! node_modules/markdown-it-pseudocodejs
npm ERR!   markdown-it-pseudocodejs@"https://github.com/samuelhorwitz/markdown-it-pseudocodejs" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /Users/jwardell1/.npm/_logs/2023-03-01T15_13_56_898Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jwardell1/.npm/_logs/2023-03-01T15_13_56_898Z-debug-0.log
```

I have installed `katex` using `npm install katex`. Is there something I'm doing wrong? I'm using mac and I would like to explore your tool.

Thank you.

**Comment by samuelhorwitz** [comment]
Hey!

So I just tried to untangle all of this old code I have up here. Basically, you aren't doing anything wrong. It's just that `markdown-it-pseudocodejs` is not available in NPM because I think I never even published it there.

I was trying to figure out where this README even came from because I didn't remember writing it, and I think I was copying the text from [markdown-it-katex](https://github.com/samuelhorwitz/markdown-it-katex) which was forked, and then just replacing things.

I know I used this in a private repository for something, but none of it was ever really polished or actually published anywhere. I probably never thought anyone would even find the repository lol.

The way I was using it in my private repository was like this, in `package.json`:

```
{
  ...
  "dependencies": {
    ...
    "markdown-it-pseudocodejs": "github:samuelhorwitz/markdown-it-pseudocodejs#master",
    ...
  }
  ...
}
```

---

*However*, I think it would be easier and safer for your own use to just go into [`index.js`](https://github.com/samuelhorwitz/markdown-it-pseudocodejs/blob/master/index.js) and utilize the code there to build your own fork.

It seems that [pseudocode.js](https://github.com/SaswatPadhi/pseudocode.js) has newer versions than whatever I was using (I had copied v1.1.0 into the repo because it seems at the time the library was not available on NPM or so I thought, but now it seems to be: https://www.npmjs.com/package/pseudocode).

In `index.js`: If you just change the first `require` to `require('pseudocode')`, and make sure install pseudocode via NPM, you can get rid of the ugly requirement to manage pseudocode.js yourself. However, it seems pseudocode.js is now in v2.x.x which means there may be breaking changes from v1.x.x which my code relies on, which might need to be debugged.

If you then just have this copy of `index.js` in your project as something like `markdown-it-pseudocodejs.js`, I think the following should just work:

```javascript
const md = require('markdown-it')();
const pseudocodePlugin = require('./markdown-it-pseudocodejs');
md.use(pseudocodePlugin);
```

The caveats are, I can't guarantee this code I have is perfect because I didn't really use it all the much, and also, I don't know if things will need further debugging because the code here is 5 years old and hasn't been checked on, while the pseudocode.js library itself has progressed into a new major version. So further cleanup may be needed. But the above should be a start.

In short, I don't recommend trying to use this code as a library but as a reference point for copying and pasting and tweaking support into your own project. Hopefully, the code can be brought over to your project with minimal effort.

**Comment by samuelhorwitz** [comment]
However, if you'd really like to use it as a library, then I recommend forking it and adding an NPM GitHub dependency: `github:jawardell/markdown-it-pseudocodejs#master`. (See also: https://docs.npmjs.com/cli/v9/configuring-npm/package-json#github-urls)

This is because relying on my repo will be risky since I can't guarantee it will remain published or published in the same place.
